### PR TITLE
feat: `vm.blobhashes`

### PR DESCRIPTION
**Pull request by onbjerg**
## Motivation

There is no way to mock blobhashes currently

## Solution

Add a cheatcode to set all blobhashes in the transaction. Note that this takes a list of blobhashes, as setting them individually by index would require us to fill the blobhash list with garbage in some cases to grow the list to an appropriate size and avoid a panic.

Closes https://github.com/foundry-rs/foundry/issues/6996

### CODE REVIEW

**Review by mattsse — APPROVED** [review]
lgtm

File: crates/cheatcodes/src/evm/fork.rs [hunk 1]
```diff
@@ -10,7 +10,6 @@ use foundry_evm_core::fork::CreateFork;
 
 impl Cheatcode for activeForkCall {
     fn apply_full<DB: DatabaseExt>(&self, ccx: &mut CheatsCtxt<DB>) -> Result {
-        let Self {} = self;
```
Comment (DaniPopes) [inline crates/cheatcodes/src/evm/fork.rs:13]: pls keep this for consistency, and it alerts you when the call changes

**Review by DaniPopes — CHANGES_REQUESTED** [review]

File: crates/cheatcodes/src/evm/fork.rs (hunk 1 above)
Comment (onbjerg) [reply crates/cheatcodes/src/evm/fork.rs:13]: what alerts you?

Comment (onbjerg) [reply crates/cheatcodes/src/evm/fork.rs]: ah the compiler, got it

**Review by Evalir — APPROVED** [review]

**Comment by onbjerg** [comment]
Gonna take a while, I have to redo the entire testdata directory since we want to run Cancun separate from every other test, and currently we just compile everything in the `testdata` directory

**Comment by flyq** [comment]
Is this pr currently blocked?

**Comment by onbjerg** [comment]
yes, but not on a feature/pr, just on time. we need to restructure most of our testing suite because we want to test cancun separately

**Comment by mattsse** [comment]
@yash-atreya this should be unblocked now with cancun specific test setup so ptal

**Comment by yash-atreya** [comment]
@mattsse ptal

**Review by mattsse — APPROVED** [review]
nice!

**Pull request merged**